### PR TITLE
update to StaticArrayInterface

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

File: src/LayoutPointers.jl [hunk 1]
```diff
@@ -20,12 +16,15 @@ using ArrayInterface:
   CPUTuple,
   static_first,
   static_step,
-  strides
+  strides,
```
Comment (ChrisRackauckas) [inline src/LayoutPointers.jl]:
```suggestion
  static_strides,
```

?

Comment (ChrisRackauckas) [reply src/LayoutPointers.jl]: You would need to update throughout the package too the things that used that.

**Comment by chriselrod** [comment]
The drop in coverage implies suggests some code that was once covered is no longer (i.e., that we're missing `static_*`).

**Comment by chriselrod** [comment]
Meh, I looked at a couple things and they seemed right.

**Pull request merged**